### ISSUE TITLE
Stage 9 Experiment Design generates invalid NLP benchmark plan and Co-Pilot chat hangs after one interaction

### ISSUE THREAD

**Issue by water5-cmd** [aiming-lab/AutoResearchClaw#253]
Thank you for building and open-sourcing AutoResearchClaw. 

I encountered two issues during Stage 09 `EXPERIMENT_DESIGN` in Co-Pilot mode.

First, the experiment design repeatedly generates an NLP/language-modeling benchmark plan even though the intended task is not an ML study. The generated `exp_plan.yaml` contains items such as:

```yaml
datasets:
  - WikiText-103

baselines:
  - LSTM LM
  - GPT-2 Small
~~~

The validation step itself reports that this generated plan is invalid. Example validation errors include:

```text
The WikiText-103 loader returns raw text examples only; there is no tokenizer, vocabulary construction, chunking/blocking, or data collator producing input_ids/labels.

Both the LSTM LM and GPT-2 baseline expect token ID tensors, so the dataset output is incompatible with the models.

DATA_CONFIG['num_classes'] is None, but get_baselines(num_classes, ...) requires a positive integer vocab size.

The custom TiedLSTMLanguageModel computes cross-entropy against labels without shifting logits/labels for causal language modeling.

The GPT-2 baseline replaces only gpt2.lm_head and sets config.vocab_size = num_classes, but does not resize or replace the input token embeddings.
```

So the generated benchmark plan appears to be both domain-mismatched and internally inconsistent.

## Second issue: Co-Pilot collaborative chat hangs

At the Stage 09 HITL gate, I selected:

```text
Action > c
```

to start the collaborative chat. The first interaction works, but when I try to continue the conversation a second time, the process hangs and becomes unusable.

This makes it difficult to correct the experiment design interactively. I can only give one round of feedback before the Co-Pilot chat freezes.

## Questions

- Is there currently a config option to disable BenchmarkAgent during Stage 09?
- Is `exp_plan.yaml` the only file used by later stages, or do I also need to edit `benchmark_plan.json` / `benchmark_agent/benchmark_plan.json`?
- Are there logs I can provide to help debug the HITL chat hang?

## Environment

```text
AutoResearchClaw version: 0.3.1
Mode: co-pilot
Environment: WSL2
Machine: Intel i9, RTX 3070, 32GB RAM
```

**Comment by Jiaaqiliu** [comment]
Hi @water5-cmd, thank you for the detailed report! Let me address both issues:

### 1. Invalid NLP Benchmark Plan for Non-ML Tasks

You can **disable BenchmarkAgent** in your config:

```yaml
experiment:
  benchmark_agent:
    enabled: false
```

BenchmarkAgent has a domain detection gate that should auto-skip non-ML tasks, but if your topic contains ML-adjacent keywords, it may classify the domain as ML and generate inappropriate benchmarks. Disabling it is the cleanest solution for non-ML research.

To answer your questions:
- **`exp_plan.yaml` is the primary file** used by later stages. `benchmark_plan.json` is supplementary and only consumed if BenchmarkAgent ran successfully.
- If you edit `exp_plan.yaml` directly (via `edit exp_plan.yaml` in co-pilot mode or manually in the run directory), later stages will use your corrected plan.

### 2. Co-Pilot Chat Hangs After One Interaction

This is a known limitation in the collaboration loop — each chat turn accumulates all prior messages plus artifact contents into the LLM context. By the second turn, the context can become very large (especially with experiment design artifacts), causing the LLM call to take significantly longer or time out. On WSL2 this effect is more pronounced.

**Workarounds:**
1. Use `edit exp_plan.yaml` to directly modify the plan instead of multi-turn chat
2. Type `done` after your first chat interaction, then re-enter co-pilot mode if needed (resets the conversation context)
3. Increase your LLM timeout: `llm: { timeout_sec: 600 }`

We'll improve the collaboration loop's context management in a future release. Thank you for the report!
